### ISSUE TITLE
Search not working properly

### ISSUE THREAD

**Issue by mendel5** [fwupd/lvfs-website#648]
Hi, I have a Lenovo Thinkpad X1 Carbon Gen7 and was using the search function on the LVFS website recently to search for devices. I noticed that the search was not working as expected, however it is not completely broken.

Here are some links to examples:
* :heavy_check_mark: Empty search page: https://fwupd.org/lvfs/search (works as expected)
* :heavy_check_mark: This search https://fwupd.org/lvfs/search?value=thinkpad+carbon returns "Showing 27 of 27 results for thinkpad,carbon:" (works as expected)
* :heavy_check_mark: This search https://fwupd.org/lvfs/search?value=thinkpad+x1+carbon returns "Showing 27 of 27 results for thinkpad,x1,carbon:" (works as expected)
* :negative_squared_cross_mark: This search https://fwupd.org/lvfs/search?value=x1+carbon+7 returns "Showing 2 of 2 results for 7,carbon:" even though I expected "x1,carbon,7"
* :negative_squared_cross_mark: This search https://fwupd.org/lvfs/search?value=thinkpad+x1+carbon+7 returns "Showing 2 of 2 results for 7,carbon:" even though I expected "thinkpad,x1,carbon,7".

As far as I can tell the search ignores some search queries (e.g. "carbon" in the last two searches above) instead of using all queries as expected.

**Comment by hughsie** [comment]
This is what we're doing: https://github.com/fwupd/lvfs-website/blob/master/lvfs/search/routes.py#L118

I think the "7" is probably too small to be a keyword. I wonder if 7th is what you want to use?

**Comment by mendel5** [comment]
You're right, if I use "7th" instead of "7" it gets recognized. Comparison:
* https://fwupd.org/lvfs/search?value=x1+carbon+7th
* https://fwupd.org/lvfs/search?value=x1+carbon+7

The reason I originally used just "7" was that when I did this search (https://fwupd.org/lvfs/search?value=thinkpad+x1+carbon) I noticed that for the X1 Carbon Gen6 there are two naming schemes:
* Gen6, e.g.  ThinkPad X1 Carbon Gen6 Thunderbolt Controller
* 6th, e.g.  ThinkPad X1 Carbon 6th Corporate ME Update

So in order to get results for both naming schemes I just wanted to use the "7" instead of a longer version like "Gen7" or "7th". Since for the Gen7 there is nothing yet named "Gen7" this doesn't matter as much as for the Gen6.

However the search keywords don't seem to be only about length:
* https://fwupd.org/lvfs/search?value=x1+carbon+7th returns "Showing 10 of 10 results for x1,carbon,7th:" (as expected)
* https://fwupd.org/lvfs/search?value=x1+carbon+gen7 returns "Showing 1 of 1 results for carbon:". This is the case even though this search keyword has one more character than the one above.

**Comment by hughsie** [comment]
I don't think Lenovo should be referring to it as "Gen7". I'll ask them what's going on.

**Comment by mendel5** [comment]
> I don't think Lenovo should be referring to it as "Gen7". I'll ask them what's going on.

To be honest I don't understand what this has to do with the naming. In my opinion it does not matter if the name is "7th" or "Gen7". Or am I missing something?

In my post above I wrote:
> However the search keywords don't seem to be only about length:
> * https://fwupd.org/lvfs/search?value=x1+carbon+7th returns "x1,carbon,7th" (as expected)
> * https://fwupd.org/lvfs/search?value=x1+carbon+gen7 returns "carbon". This is the case even though this search keyword has one more character than the one above.

Do you see the issue here that the search recognizes "7th" but not "gen7"? I think this is more about the search function on the LVFS website than about Lenovo's naming.
